pkg/common/configs/configvalidator.go

```diff
@@ -357,13 +357,32 @@ func checkLimit(limit Limit, currIdx int, userWildCardIdx, groupWildCardIdx *int
 	}
 	// at least some resource should be not null
 	if limit.MaxApplications == 0 && resources.IsZero(limitResource) {
-		return fmt.Errorf("invalid resource combination for limit names '%s' all resource limits are null", limit.Users)
+		return fmt.Errorf("invalid resource combination for limit %s all resource limits are null", limit.Limit)
 	}
+
+	if queue.MaxApplications != 0 && (limit.MaxApplications > queue.MaxApplications || limit.MaxApplications == 0) {
+		return fmt.Errorf("invalid MaxApplications settings for limit %s exeecd current the queue MaxApplications", limit.Limit)
```

Comment:
In the website, the limits description is following.
```
maxapplications is an unsigned integer value, larger than 1, which allows you to limit the number of running applications for the configured user or group. Specifying a zero maximum applications limit is not allowed as it would implicitly deny access. Denying access must be handled via the ACL entries.
```

so I think that the error description is not suitable when conditions (queue.MaxApplications != 0 && limit.MaxApplications == 0) happens.
I think this part will be like this.

if queue.MaxApplications != 0 {
  if limit.MaxApplications > queue.MaxApplications  {
    return fmt.Errorf("invalid MaxApplications settings for limit %s exeecd current the queue MaxApplications", limit.Limit)
  }
  if limit.MaxApplications == 0 {
    return fmt.Errorf("MaxApplications is 0 in limit name %s, it should be 1 ~ %d", limit.Limit, queue.MaxApplications)
  }
}

Reply:
Hi @0yukali0 , thank you for review, but if we don't set the MaxApplications, it will have a default value, it will be 0. 
If we want to change to init the MaxApplications to an unlimited value rather than 0?

Reply:
No,  the default value of MaxApplications in the limit is 0.
The message hints that admin should set the MaxApplications in the limit when the MaxApplications in the queue is set.
Because users will not know that setting the MaxApplications in limit is necessary after setting the MaxApplications in the queue.

Reply:
Good catch @0yukali0 , it makes sense to me now!

Reply:
Addressed in latest PR now.

Reply:
Hi @zhuqi-lucas, I will merge this to the master.
Thanks for  the contribution.